docs/3.api/2.composables/on-prehydrate.md

```diff
@@ -0,0 +1,61 @@
+---
+title: "onPrehydrate"
+description: "Use onPrehydrate to run a callback on the client immediately before
+Nuxt hydrates the page."
+links:
+  - label: Source
+    icon: i-simple-icons-github
+    to: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/composables/ssr.ts
+    size: xs
+---
+
+::important
+This composable will be available in Nuxt v3.12+.
+::
+
+`onPrehydrate` is a composable lifecycle hook that allows you to run a callback on the client immediately before
+Nuxt hydrates the page.
+
+::note
+This is an advanced utility and should be used with care.
+::
+
+## Usage
+
+`onPrehydrate` can be called directly in the setup function of a Vue component, or in a plugin. It will only have an
```

Comment:
"setup function"  also refers to "inside `<script setup>`" but I see the risk that this might not be clear for people not aware of `setup()` without the syntactic sugar